server/src/test/java/org/apache/druid/metadata/IndexerSQLMetadataStorageCoordinatorTest.java

```diff
@@ -848,6 +848,28 @@ public void testDeleteDataSourceMetadata() throws IOException
     Assert.assertNull("getDataSourceMetadataNullAfterDelete", coordinator.retrieveDataSourceMetadata("fooDataSource"));
   }
 
+  @Test
+  public void testDeleteSegments()
+  {
+    Exception e = null;
+    try {
+      coordinator.deleteSegments(SEGMENTS);
+    }
+    catch (Exception ex) {
+      e = ex;
+    }
+    Assert.assertNull(e);
+    Assert.assertEquals(
+            0,
+            ImmutableSet.copyOf(
+                    coordinator.retrieveUnusedSegmentsForInterval(
+                            defaultSegment.getDataSource(),
+                            defaultSegment.getInterval()
+                    )
+            ).size()
+    );
+  }
```

Comment:
I don't understand what this is testing. Could you explain what you're trying to test here. You could use the test method name format like `test<MethodUnderTest>When<Condition>Should<ExpectedResult>`

See `KafkaRecordSupplierTest` for some examples like `getLatestSequenceNumberWhenPartitionIsEmptyAndUseEarliestOffsetShouldReturnsValidNonNull`

Reply:
Sure, I did a little change. What I want to test is the function of deleting segment metadata in MetaDataStorage.
Here is the workflow of this UT: `Published segments to MetaDataStorage` -> `check segments Published` -> `remove segments in MetaDataStorage` -> `check segments removed`.
Also name have been changed as `testDeleteSegmentsInMetaDataStorage`